### PR TITLE
mon/MgrMonitor: send digests only if is_active()

### PR DESCRIPTION
**Pull request by tchaikov**
labeling this a bug fix, because we should cancel an event before resetting it, otherwise the event is leaked.

### CODE REVIEW

File: src/mon/MgrMonitor.cc [hunk 1]
```diff
@@ -346,6 +337,12 @@ void MgrMonitor::tick()
     return;
 
   const auto now = ceph::coarse_mono_clock::now();
+  if (is_active() &&
+      chrono::duration_cast<chrono::seconds>(now - last_sent_digest).count() >
+      g_conf->mon_mgr_digest_period) {
```
Comment (jcsp) [inline src/mon/MgrMonitor.cc]: Hmm, this is going to give pretty irregular messages if mon_mgr_digest_period is not a multiple of the tick period.

**Comment by tchaikov** [comment]
@jcsp makes sense! fixed and repushed.

**Comment by liewegas** [comment]
<pre>
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/Monitor.cc:77:0:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/MgrMonitor.h: In constructor ‘MgrMonitor::MgrMonitor(Monitor*, Paxos*, const string&)’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/MgrMonitor.h:46:42: error: class ‘MgrMonitor’ does not have any field named ‘digest_callback’
     : PaxosService(mn, p, service_name), digest_callback(nullptr)
</pre>

**Comment by tchaikov** [comment]
@liewegas fixed and pushed.

**Pull request merged**